### PR TITLE
[DEVOPS-1578] Automate QA deploy messages

### PR DESCRIPTION
**Pull request by michalchecinski**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/DEVOPS-1578

## 🚧 Type of change

-   🚀 New feature development
-   🤖 Build/deploy pipeline (DevOps)

## 📔 Objective

Add github action that can be used across our repos to report deployment status to Slack.

## 📋 Code changes

-   **.github/workflows/test-report-deployment-status-to-slack.yml:** Workflow to test the action
- **report-deployment-status-to-slack/action.yml** Composite action that's sending Slack message about deployment status.

## ⏰ Reminders before review

-   Contributor guidelines followed
-   All formatters and local linters executed and passed
-   Written new unit and / or integration tests where applicable
-   Used internationalization (i18n) for all UI strings
-   CI builds passed
-   Communicated to DevOps any deployment requirements
-   Updated any necessary documentation or informed the documentation team

## 🦮 Reviewer guidelines

-   👍 (`:+1:`) or similar for great changes
-   📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
-   ❓ (`:question:`) for questions
-   🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
-   🎨 (`:art:`) for suggestions / improvements
-   ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
-   🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
-   ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

**Comment by Eeebru** [comment]
Maybe change the name of the action folder to `report-deployment-status-to-slack`

**Comment by michalchecinski** [comment]
@mimartin12 I wanted to keep it consistent across `gh-actions` repo. I could quickly revert to a reusable workflow if needed (have the file on hand).

**Comment by michalchecinski** [comment]
@Eeebru I think the name of the folder is the same as yours. But maybe my eyesight is playing tricks on me 😅

**Comment by Eeebru** [comment]
> * Build/deploy pipeline (DevOps)

@michalchecinski  `test-report-deployment-status-to-slack` -> `report-deployment-status-to-slack`

**Comment by michalchecinski** [comment]
@Eeebru Ah, okay. But this is a test of the workflow, so I followed our naming from other like  `test-download-artifacts.yml` or `test-release-version-check.yml`.

File: .github/workflows/test-report-deployment-status-to-slack.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    uses: ./.github/workflows/_report_status_to_slack.yml
+    with:
+      environment: US QA Cloud
+      tag: test-workflow
+      slack-channel: devops-alerts
+      failure: false
+      AZURE_KV_CI_SERVICE_PRINCIPAL: ${{ secrets.AZURE_KV_CI_SERVICE_PRINCIPAL }}
+
+  test-report-failure:
+    uses: ./.github/workflows/_report_status_to_slack.yml
```
Comment (mimartin12) [inline .github/workflows/test-report-deployment-status-to-slack.yml:20]: This path doesn't seem to exist. `report-deployment-status-to-slack/` might just need an underscore.

Comment (michalchecinski) [reply .github/workflows/test-report-deployment-status-to-slack.yml]: Thanks for noticing it!

**Review by mimartin12 — APPROVED** [review]

**Pull request merged**